android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/RepositoryServiceConnection.java

```diff
@@ -61,13 +62,21 @@ public RepositoryServiceConnection(Context context, ConnectionLifeCycle connecti
   }
 
   private void walletNotInstalledBehaviour() {
-    onServiceConnected(new ComponentName("", ""), new IBinderWalletNotInstalled());
+    onServiceConnected(new ComponentName("", AppcoinsBillingStubHelper.class.getSimpleName()),
+        new IBinderWalletNotInstalled());
+  }
+
+  private void bindFailedBehaviour() {
+    onServiceConnected(new ComponentName("", UriCommunicationAppcoinsBilling.class.getSimpleName()),
+        new IBinderWalletNotInstalled());
   }
 
   private void walletInstalledBehaviour(String packageName) {
     String iabAction = WalletUtils.getIabAction();
     Intent serviceIntent = new Intent(iabAction);
     serviceIntent.setPackage(packageName);
-    context.bindService(serviceIntent, this, Context.BIND_AUTO_CREATE);
+    if (context.bindService(serviceIntent, this, Context.BIND_AUTO_CREATE)) {
```

Comment:
I believe this condition has to be inverted. if(!context.bindService...)

Reply:
you are right, my bad :man_facepalming: